Internacionalization support (English and Portuguese)

ArthurMota9 opened the pull request:
# Summary

#226 

Currently [app.brmodeloweb.com](app.brmodeloweb.com) is only available in pt-br and have no support to introduce other languages to the system easily. In order to increase our reach and allow monetization possibilities with [Carbonads.net](https://www.carbonads.net/) we need to to translate the app to English.

Configuration with `angular-translate` was added earlier in [this PR](https://github.com/brmodeloweb/brmodelo-app/pull/152), so it was necessary to "only" include the translations in all files and save the option selected by the user in localStorage. If there is no value in localStorage, the browser language is considered, being `pt-br` or `en`.

At some point in the future we will probably allow editing of user information (name, email), so we can include the language in the database and remove it from local storage.

# List of changes
- Add pt_BR and en translations to JS and HTML files;
- Add chosen translation to `localStorage`
- Get language by browser.

ArthurMota9 committed:
Add pt_BR translations

ArthurMota9 committed:
Add english translations

ArthurMota9 committed:
Add translations to html files

ArthurMota9 committed:
Add translations to js files

ArthurMota9 committed:
Add translation chosen in localStorage

ArthurMota9 committed:
Add translations to forgotten placeholders

ArthurMota9 committed:
Add translations to conversorService

ArthurMota9 committed:
Add logout translation

ArthurMota9 committed:
Merge branch 'development' into feature/add-translations

oicramps commented on app/angular/components/conversionOptionModal.html:
```diff
@@ -16,5 +16,5 @@ <h3 class="modal-title">{{$ctrl.title}}</h3>
 
 <div class="modal-footer">
-	<button type="button" class="br-button" ng-click="$ctrl.forward($ctrl.selectedValue)">Avançar</button>
+	<button type="button" class="br-button" ng-click="$ctrl.forward($ctrl.selectedValue)">{{ 'Advance' | translate }}</button>
```
```suggestion
	<button type="button" class="br-button" ng-click="$ctrl.forward($ctrl.selectedValue)">{{ 'Forward' | translate }}</button>
```
or maybe?
```suggestion
	<button type="button" class="br-button" ng-click="$ctrl.forward($ctrl.selectedValue)">{{ 'Next' | translate }}</button>
```

oicramps commented on app/angular/components/deleteModelModal.html:
```diff
@@ -1,12 +1,12 @@
 
 <div class="modal-body">
-	<p>Tem certeza que deseja <strong>excluir</strong> a modelagem?</p>
+	<p>{{ 'Are you sure you want to delete the model?' | translate }}</p>
```
We won't have an emphasis on the delete word anymore?

ArthurMota9 replied:
For the translations to work including html elements we have to include some configs, I'll do it later.

oicramps commented on app/angular/conceptual/conceptual.html:
```diff
@@ -54,7 +54,7 @@ <h2 class="h4 pull-left">Modelo conceitual de: {{$ctrl.model.name}}</h2>
 	  <sidebar-control-conceptual selected="$ctrl.selectedElement" on-update="$ctrl.onUpdate(event)"></sidebar-control-conceptual>
 
-	<a class="br-help hidden-print" href="https://github.com/brmodeloweb/brmodelo-app/issues/new/choose" target="_blank" rel="noopener noreferrer" title="Reportar bug ou requisitar funcionalidade">
+	<a class="br-help hidden-print" href="https://github.com/brmodeloweb/brmodelo-app/issues/new/choose" target="_blank" rel="noopener noreferrer" title="{{ 'Report bug or request functionality' | translate }}">
```
```suggestion
	<a class="br-help hidden-print" href="https://github.com/brmodeloweb/brmodelo-app/issues/new/choose" target="_blank" rel="noopener noreferrer" title="{{ 'Report a bug or request a feature' | translate }}">
```

oicramps commented on app/angular/conceptual/conceptual.js:
```diff
@@ -107,7 +107,7 @@ const controller = function (ModelAPI, $stateParams, $rootScope, $timeout, $uibM
 			controller: function () {
 				const $ctrl = this;
-				$ctrl.suggestedName = `${model.name} (cópia)`;
+				$ctrl.suggestedName = $filter('translate')('X (copy)', { name: model.name });
```
```suggestion
				$ctrl.suggestedName = $filter('translate')('modelName (copy)', { name: model.name });
```

I think it is more descriptive if we use a meaningful word that describes the data placed there instead of using `X`.

oicramps commented on app/angular/logic/logic.html:
```diff
@@ -30,19 +30,19 @@
 			<header class="workspace-header">
 				<div class="page-header clearfix">
-					<h2 class="h4 pull-left">Modelo lógico de: {{$ctrl.model.name}}</h2>
+					<h2 class="h4 pull-left">{{ 'Logic model of:' | translate }} {{$ctrl.model.name}}</h2>
```
```suggestion
					<h2 class="h4 pull-left">{{ 'Logical model of:' | translate }} {{$ctrl.model.name}}</h2>
```

oicramps commented on app/angular/logic/logic.js:
```diff
@@ -171,7 +172,7 @@ const controller = function (
 	ctrl.editColumn = function (oldColumn, editedColumn, $index) {
 		if (editedColumn.name == "") {
-			ctrl.showFeedback("NOME de coluna não pode ficar em branco!", true, "error");
+			ctrl.showFeedback("Column name cannot be empty!", true, "error");
```
```suggestion
			ctrl.showFeedback("The column name cannot be empty!", true, "error");
```

oicramps commented on app/angular/logic/logic.js:
```diff
@@ -189,12 +190,12 @@ const controller = function (
 	ctrl.addColumn = function (column) {
 		if (column.name == "") {
-			ctrl.showFeedback("NOME de coluna não pode ficar em branco!", true, "error");
+			ctrl.showFeedback("Column name cannot be empty!", true, "error");
```
```suggestion
			ctrl.showFeedback("The column name cannot be empty!", true, "error");
```

oicramps commented on app/angular/logic/logic.js:
```diff
@@ -189,12 +190,12 @@ const controller = function (
 
 		if (column.FK && column.tableOrigin.idName == "") {
-			ctrl.showFeedback("Selecione a origem da tabela estrangeira!", true, "error");
+			ctrl.showFeedback("Select source of foreign table!", true, "error");
```
```suggestion
			ctrl.showFeedback("Select the source of the foreign table!", true, "error");
```

idmarjr replied:
I would simplify it a bit: 
> _Select the foreign table source!_

oicramps commented on app/angular/login/login.html:
```diff
@@ -47,19 +47,19 @@ <h1>BR Modelo Web</h1>
+		</footer>
 	</div>
-	<a class="br-help hidden-print" href="https://github.com/brmodeloweb/brmodelo-app/issues/new/choose" target="_blank" rel="noopener noreferrer" title="Reportar bug ou requisitar funcionalidade">
+	<a class="br-help hidden-print" href="https://github.com/brmodeloweb/brmodelo-app/issues/new/choose" target="_blank" rel="noopener noreferrer" title="{{ 'Report bug or request functionality' | translate }}">
```
```suggestion
	<a class="br-help hidden-print" href="https://github.com/brmodeloweb/brmodelo-app/issues/new/choose" target="_blank" rel="noopener noreferrer" title="{{ 'Report a bug or request a feature' | translate }}">
```

oicramps commented on app/angular/recovery/recovery.html:
```diff
@@ -8,36 +8,36 @@
 		</section>
 	</div>
-	<a class="br-help hidden-print" href="https://github.com/brmodeloweb/brmodelo-app/issues/new/choose" target="_blank" rel="noopener noreferrer" title="Reportar bug ou requisitar funcionalidade">
+	<a class="br-help hidden-print" href="https://github.com/brmodeloweb/brmodelo-app/issues/new/choose" target="_blank" rel="noopener noreferrer" title="{{ 'Report bug or request functionality' | translate }}">
```
```suggestion
	<a class="br-help hidden-print" href="https://github.com/brmodeloweb/brmodelo-app/issues/new/choose" target="_blank" rel="noopener noreferrer" title="{{ 'Report a bug or request a feature' | translate }}">
```

idmarjr replied:
In another subject: We should definitively create a component for this bug report thing

idmarjr replied:
Work in progress at #266

oicramps commented on app/angular/recovery/recovery.js:
```diff
@@ -13,23 +13,23 @@ const controller = function (AuthService) {
 	const handleSuccess = () => {
 		ctrl.loading = false;
-		showFeedback(`Um email com instruções foi enviado para ${ctrl.mail}`, "success");
+		showFeedback($filter('translate')('An email with instructions has been sent to X', { email: ctrl.mail }), "success");
```
```suggestion
		showFeedback($filter('translate')('An email with instructions has been sent to EMAIL', { email: ctrl.mail }), "success");
```

Again, I think we could have a meaningful word instead of just `X`

oicramps commented on app/angular/recovery/reset.html:
```diff
@@ -30,30 +30,30 @@ <h1>Redefinir senha</h1>
 		</section>
 	</div>
-	<a class="br-help hidden-print" href="https://github.com/brmodeloweb/brmodelo-app/issues/new/choose" target="_blank" rel="noopener noreferrer" title="Reportar bug ou requisitar funcionalidade">
+	<a class="br-help hidden-print" href="https://github.com/brmodeloweb/brmodelo-app/issues/new/choose" target="_blank" rel="noopener noreferrer" title="{{ 'Report bug or request functionality' | translate }}">
```
```suggestion
	<a class="br-help hidden-print" href="https://github.com/brmodeloweb/brmodelo-app/issues/new/choose" target="_blank" rel="noopener noreferrer" title="{{ 'Report a bug or request a feature' | translate }}">
```

oicramps commented on app/angular/recovery/reset.js:
```diff
@@ -38,11 +38,11 @@ const controller = function (AuthService, $stateParams, $timeout, $state) {
 
 	const handleSuccess = () => {
-		showFeedback(`Nova senha salva com sucesso.`, "success");
+		showFeedback("New password saved successfully.", "success");
```
```suggestion
		showFeedback("The new password was successfully saved.", "success");
```

oicramps commented on app/angular/recovery/reset.js:
```diff
@@ -63,14 +63,14 @@ const controller = function (AuthService, $stateParams, $timeout, $state) {
+				showFeedback("The password must be the same in both fields.");
 			} else if (newPassword.length < 6) {
-				showFeedback("Senha deve conter mais que 6 dígitos");
+				showFeedback("Password must contain more than 6 digits.");
```
```suggestion
				showFeedback("The password must contain more than 6 digits.");
```

oicramps commented on app/angular/service/conversorService.js:
```diff
@@ -75,12 +75,12 @@ const logicConversorService = (ConceptualService, $uibModal) => {
-							$ctrl.title = `Assistente de conversão - Especialização`;
-							$ctrl.summary = `O que deseja fazer com a herança partindo da tabela ${getExtensionRootName(extension)}?`;
+							$ctrl.title = $filter('translate')("Conversion Assistant - Specialization");
+							$ctrl.summary = $filter('translate')("What do you want to do with inheritance from table X?", { table: getExtensionRootName(extension) });
```
```suggestion
							$ctrl.summary = $filter('translate')("What do you want to do with inheritance from table TABLE_NAME?", { table: getExtensionRootName(extension) });
```

A meaningful word instead of just `X`

oicramps commented on app/angular/service/conversorService.js:
```diff
@@ -139,11 +139,11 @@ const logicConversorService = (ConceptualService, $uibModal) => {
-						$ctrl.title = `Atenção - Assistente de conversão - Especialização`;
-						$ctrl.summary = `Não foi possível realizar essa conversão pois a tabela ${getExtensionRootName(extension)} possui outras conexões. Escolha outra opção`;
+						$ctrl.title = $filter('translate')("Attention - Conversion Assistant - Specialization");
+						$ctrl.summary = $filter('translate')("It was not possible to perform this conversion because table X has other connections. Choose another option.", { table: getExtensionRootName(extension) });
```
```suggestion
						$ctrl.summary = $filter('translate')("It was not possible to perform this conversion because table TABLE_NAME has other connections. Choose another option.", { table: getExtensionRootName(extension) });
```

oicramps commented on app/angular/service/conversorService.js:
```diff
@@ -312,11 +312,11 @@ const logicConversorService = (ConceptualService, $uibModal) => {
-									$ctrl.title = `Assistente de conversão: Atributo multivalorado`;
-									$ctrl.summary = `O que deseja fazer com atributo multivalorado ${attName} da tabela ${table.name} ?`;
+									$ctrl.title = $filter('translate')("Conversion Assistant: Multivalued Attribute");
+									$ctrl.summary = $filter('translate')("What do you want to do with multivalued attribute X of table Y?", { attribute: attName, table: table.name });
```
```suggestion
									$ctrl.summary = $filter('translate')("What do you want to do with multivalued attribute ATTR_NAME of table TABLE_NAME?", { attribute: attName, table: table.name });
```

oicramps commented on app/angular/signup/signup.html:
```diff
@@ -8,64 +8,64 @@
 		</section>
 	</div>
-	<a class="br-help hidden-print" href="https://github.com/brmodeloweb/brmodelo-app/issues/new/choose" target="_blank" rel="noopener noreferrer" title="Reportar bug ou requisitar funcionalidade">
+	<a class="br-help hidden-print" href="https://github.com/brmodeloweb/brmodelo-app/issues/new/choose" target="_blank" rel="noopener noreferrer" title="{{ 'Report bug or request functionality' | translate }}">
```
```suggestion
	<a class="br-help hidden-print" href="https://github.com/brmodeloweb/brmodelo-app/issues/new/choose" target="_blank" rel="noopener noreferrer" title="{{ 'Report a bug or request a feature' | translate }}">
```

oicramps commented on app/angular/workspace/workspace.html:
```diff
@@ -36,11 +36,11 @@ <h2 class="h2 pull-left">Modelagens</h2>
+							<th><strong>{{ 'Type' | translate }}</strong></th>
+							<th><strong>{{ 'Name' | translate }}</strong></th>
+							<th><strong>{{ 'Author' | translate }}</strong></th>
+							<th><strong>{{ 'Creation' | translate }}</strong></th>
```
```suggestion
							<th><strong>{{ 'Created at' | translate }}</strong></th>
```

oicramps commented on app/angular/workspace/workspace.html:
```diff
@@ -50,17 +50,17 @@ <h2 class="h2 pull-left">Modelagens</h2>
 		</div><!-- End .container.projectsList -->
 
-		<a class="br-help hidden-print" href="https://github.com/brmodeloweb/brmodelo-app/issues/new/choose" target="_blank" rel="noopener noreferrer" title="Reportar bug ou requisitar funcionalidade">
+		<a class="br-help hidden-print" href="https://github.com/brmodeloweb/brmodelo-app/issues/new/choose" target="_blank" rel="noopener noreferrer" title="{{ 'Report bug or request functionality' | translate }}">
```
```suggestion
		<a class="br-help hidden-print" href="https://github.com/brmodeloweb/brmodelo-app/issues/new/choose" target="_blank" rel="noopener noreferrer" title="{{ 'Report a bug or request a feature' | translate }}">
```

ArthurMota9 committed:
Fix translations

ArthurMota9 committed:
Merge branch 'development' into feature/add-translations

ArthurMota9 committed:
Fix zoom translations

ArthurMota9 committed:
Fix tests and add a new one

Adds test that checks if clicking on the flags on the login screen changes
the language

ArthurMota9 commented on cypress/integration/login.spec.js:
```diff
@@ -53,15 +53,27 @@ describe('Login', () => {
+        .should('be.visible')
+    })
+
+    it('change language', () => {
```
@wlsf82 Can you validate if everything is ok with this test?

wlsf82 replied:
@ArthurMota9, it looks good to me.
Did you run them? Are they still passing?

By the way, soon, I intend to work on having them running on CI.

ArthurMota9 replied:
All tests passed, I just mentioned you to verify that the code respects the standard. Thanks!

idmarjr committed:
Merge branch 'development' into feature/add-translations

idmarjr committed:
Improve translation: Created

Small translation change

Where: Workspace (List of models)
What: "Created at" --> Created

idmarjr committed:
Improve translations: Rename / Duplicate / Delete

Where: Workspace
What: Remove word "model" from "Rename model" and "Duplicate model"

This way we can keep a generic action and reuse this translation in other places if needed

idmarjr committed:
Improve translation: Modelo -> Modelagem

idmarjr committed:
Remove empty line at the bottom of the file

oicramps reviewed (approved):
LGTM! Excellent job, @ArthurMota9  🚀  👏
